usaspending_api/recipient/v2/views.py

```diff
@@ -0,0 +1,116 @@
+import logging
+from datetime import datetime
+from dateutil.relativedelta import relativedelta
+from collections import OrderedDict
+
+from rest_framework.response import Response
+from usaspending_api.common.views import APIDocumentationView
+
+from django.db.models import Sum
+
+from usaspending_api.awards.models_matviews import UniversalTransactionView
+from usaspending_api.awards.v2.filters.filter_helpers import sum_transaction_amount
+# from usaspending_api.awards.v2.filters.sub_award import subaward_filter
+from usaspending_api.awards.v2.filters.transaction import transaction_filter
+from usaspending_api.awards.v2.filters.matview_filters import universal_transaction_matview_filter
+from usaspending_api.common.exceptions import InvalidParameterException
+from usaspending_api.common.helpers import table_exists, generate_fiscal_year
+from usaspending_api.recipient.models import StateData
+
+logger = logging.getLogger(__name__)
+
+
+class StateMetaDataViewSet(APIDocumentationView):
+
+    def get_state_data(self, state_data_results, field, year=None):
+        """Finds which earliest or latest state data to use based on the year and what data is available"""
+        state_data = OrderedDict(sorted([(str(state_data['year']), state_data) for state_data in state_data_results
+                                         if state_data[field]], key=lambda pair: pair[0]))
+        earliest = list(state_data.keys())[0]
+        latest = list(state_data.keys())[-1]
+        if year and year.isdigit() and year < earliest:
+            return state_data[earliest]
+        elif year and year.isdigit() and earliest <= year <= latest:
+            return state_data[year]
+        elif not year or year > latest or year == 'all' or year == 'latest':
+            return state_data[latest]
+        else:
+            raise InvalidParameterException('Invalid year: {}.'.format(year))
+
+    def get(self, request, fips):
```

Comment:
Agreed, fixed.

Reply:
Just to make a note of it (not a problem right now since there aren't any 3 character FIPS codes), if we ever get into a situation where any submission or data source always pads the code to a certain length, for example 3 characters, then we need to support things like `002`, `02, `2` since they are technically equivalent "numbers". 

Not something we need to handle right now, but just a thought.